### PR TITLE
Add Mellanox ConnectX4 Lx EN and ConnectX5 dpdk examples

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -336,6 +337,13 @@ The allocated device information are exported in Container's environment variabl
 For example, if 2 devices are allocated from `intel.com/sriov` extended resource then the allocated device information will be found in following env variable:
 `PCIDEVICE_INTEL_COM_SRIOV=0000:03:02.1,0000:03:04.3`
 
+### Mellanox Rdma Dpdk
```
Comment (moshe010) [inline README.md]: s/Rdma/RDMA
s/Dpdk/DPDK

File: README.md [hunk 2]
```diff
@@ -336,6 +337,13 @@ The allocated device information are exported in Container's environment variabl
...
+
+Sriov device plugin is integerated with mellanox rdma, the following example shows how to get a pod with 2 mellanox VFs with rdma dpdk usign 1G hugepages:
```
Comment (moshe010) [inline README.md]: Sriov/SR-IOV

Comment (moshe010) [inline README.md]: s/rdma/RDMA
s/dpdk/DPDK

File: deployments/pod-mlnx-dpdk.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: mlnx-dpdk
+  annotations:
+    k8s.v1.cni.cncf.io/networks: sriov-net1, sriov-net1
```
Comment (moshe010) [inline deployments/pod-mlnx-dpdk.yaml]: let do a simple that we add 1 network, so replace "k8s.v1.cni.cncf.io/networks: sriov-net1, sriov-net1"
with k8s.v1.cni.cncf.io/networks: sriov-net1

File: deployments/pod-mlnx-dpdk.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+spec:
+  securityContext:
```
Comment (moshe010) [inline deployments/pod-mlnx-dpdk.yaml]: we don't need Line 8-9

File: deployments/pod-mlnx-dpdk.yaml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    runAsUser: 0
+  containers:
+  - name: mlnx-dpdk
+    image: mmdh/mlnx-dpdk
+    imagePullPolicy: IfNotPresent
+    securityContext:
+      capabilities:
+        add: ["IPC_LOCK", "NET_RAW", "NET_ADMIN"]
```
Comment (moshe010) [inline deployments/pod-mlnx-dpdk.yaml]: you can remove NET_ADMIN

File: deployments/pod-mlnx-dpdk.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    command: [ "/bin/bash", "-c", "--" ]
+    args: [ "while true; do sleep 300000; done;" ]
+    resources:
+      requests:
+        intel.com/sriov: '2' 
+        memory: 1000Mi
+        hugepages-1Gi: 8Gi
+        cpu: '8'
+      limits:
+        intel.com/sriov: '2'
```
Comment (moshe010) [inline deployments/pod-mlnx-dpdk.yaml]: change to mellanox.com/sriov: '1'

File: deployments/pod-mlnx-dpdk.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        hugepages-1Gi: 8Gi
+        cpu: '8'
+        memory: 1000Mi
+    volumeMounts:
+    - mountPath: /mnt/huge
+      name: hugepage
+      readOnly: False
+    - mountPath: /var
```
Comment (moshe010) [inline deployments/pod-mlnx-dpdk.yaml]: you don't need to mount /var

File: README.md (hunk 2 above)
Comment (Mmduh-483) [reply README.md]: fixed

Comment (Mmduh-483) [reply README.md]: fixed

File: deployments/pod-mlnx-dpdk.yaml (hunk 3 above)
Comment (Mmduh-483) [reply deployments/pod-mlnx-dpdk.yaml]: fixed

File: deployments/pod-mlnx-dpdk.yaml (hunk 4 above)
Comment (Mmduh-483) [reply deployments/pod-mlnx-dpdk.yaml]: fixed

**Comment by Mmduh-483** [comment]
@Mmduh-483 the previous comments are fixed now

**Comment by zshi-redhat** [comment]
@Mmduh-483 @moshe010 shall we separate RDMA and DPDK into different sections and have this one focus on RMDA?  I feel it's worth a separate section for RDMA as it's a new feature than DPDK (although the can be used together).
For DPDK, we probably want to have two examples, one for DPDK with kernel interface(e.g Mellanox card), the other for DPDK with vfio-pci driver (e.g Intel card), wdyt?

**Comment by Mmduh-483** [comment]
@zshi-redhat I updated the patch to be only for dpdk also added examples, I will create an new PR for the RDMA

**Comment by Mmduh-483** [comment]
@adrianchiris

File: docs/dpdk/mlnx-dpdk/crd-dpdk-connectx5.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-dpdk-mlnx-connectx5
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: intel.com/mlnx_connectx5
+spec:
+  config: '{
+    "type": "sriov",
+    "name": "sriov-mlnx_connectx5",
+    "vlan": 2000
```
Comment (adrianchiris) [inline docs/dpdk/mlnx-dpdk/crd-dpdk-connectx5.yaml]: @Mmduh-483 , do we really need to specify vlan here ?

File: docs/dpdk/mlnx-dpdk/crd-dpdk-connectx4.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+kind: NetworkAttachmentDefinition
+metadata:
+  name: sriov-dpdk-mlnx-connectx4
+  annotations:
+    k8s.v1.cni.cncf.io/resourceName: intel.com/mlnx_connectx4
+spec:
+  config: '{
+    "type": "sriov",
+    "name": "sriov-mlnx_connectx4",
+    "vlan": 2000
```
Comment (adrianchiris) [inline docs/dpdk/mlnx-dpdk/crd-dpdk-connectx4.yaml]: @Mmduh-483 , do we really need to specify vlan here ?

File: docs/dpdk/README.md [hunk 10]
```diff
@@ -15,6 +15,8 @@ For more information on hugepage support in Kubernetes please see [here](https:/
 ### VF drivers
 DPDK applications require devices to be attached with supported dpdk backend driver.
 For Intel® x700 series NICs `igb_uio` or `vfio-pci` is required.
+For Mellanox ConnectX®-4 Lx EN Adapter `mlx5_core` or `mlx5_ib` is required.
+For Mellanox ConnectX®-5 Adapter `mlx5_core` or `mlx5_ib` is required.
```
Comment (adrianchiris) [inline docs/dpdk/README.md]: perhaps make these bullets and put both connectX4 and 5 in the same bullet e.g:

### VF drivers
DPDK applications require devices to be attached with supported dpdk backend driver.
- For Intel® x700 series NICs `igb_uio` or `vfio-pci` is required.	For Intel® x700 series NICs `igb_uio` or `vfio-pci` is required.
- For Mellanox ConnectX®-4 Lx,  ConnectX®-5 Adapters `mlx5_core` or `mlx5_ib` is required.

**Comment by Mmduh-483** [comment]
> /lgtm
> 
> nit: The resource name in network crd is using `intel.com`, perhaps worth a note indicating it's the default resource name prefix if not explicitly specified in cmdline parameter. but this doesn't affect the usability.

@zshi-redhat this is already mentioned in the main Readme Command line arguments
 section, are there other files that need to add it?

File: docs/dpdk/mlnx-dpdk/crd-dpdk-connectx4.yaml (hunk 9 above)
Comment (Mmduh-483) [reply docs/dpdk/mlnx-dpdk/crd-dpdk-connectx4.yaml]: no

**Comment by zshi-redhat** [comment]
> > /lgtm
> > nit: The resource name in network crd is using `intel.com`, perhaps worth a note indicating it's the default resource name prefix if not explicitly specified in cmdline parameter. but this doesn't affect the usability.
> 
> @zshi-redhat this is already mentioned in the main Readme Command line arguments
> section, are there other files that need to add it?

no, we should be good now.

**Comment by Mmduh-483** [comment]
@zshi-redhat any other comments?

**Comment by zshi-redhat** [comment]
> @zshi-redhat any other comments?

Hi @Mmduh-483 this looks good to me, I don't have any other comments.

**Pull request merged**